thirdparty/openssl/CMakeLists.txt

```diff
@@ -27,7 +26,23 @@ if(CHOST)
     list(APPEND CFG_ENV_VAR CROSS_COMPILE=${CHOST}-)
 endif()
 
-set(CFG_OPTS shared)
+if(MONOLIBTIC)
+    set(CFG_OPTS no-shared)
+else()
+    set(CFG_OPTS shared)
+endif()
+
+list(APPEND CFG_OPTS
+    no-comp
+    no-default-thread-pool
+    no-hw
+    no-legacy
+    no-quic
+    no-thread-pool
+    no-ui-console
+    no-uplink
+    no-whirlpool
```

Comment:
I had commented on the commit, so replicating it here so I can actually find it again in the future.

---

I see, `no-legacy` is about the provider only but not about the algorithms. From the name I incorrectly assumed it covered all of these:

>   * Moved all variations of the EVP ciphers CAST5, BF, IDEA, SEED, RC2,
    RC4, RC5, and DES to the legacy provider.
>   * Moved the EVP digests MD2, MD4, MDC2, WHIRLPOOL and RIPEMD-160 to the legacy
    provider.

But presumably that does provide a (incomplete?) list of various others to disable.